integrity disk

wibarre opened the pull request:
<!--
1. Please remember to include a DCO on every commit (`git commit -s`)
https://github.com/apps/dco
-->

**Pull request check list**

- [ ] Proper tests/regressions included?
- [ ] Documentation updated?

**Affected functionality**
<!-- Please provide a description of the affected functionality -->

**Description of change**
<!-- Please provide a description of the change -->

**Which issue this pull requests fixes**
<!-- optional. `fixes #<issue number>` format will close an issue when this pull request is merged -->

Victorblsilveira committed:
feat: SPIKE - Demonstrating types convertion

Victorblsilveira committed:
feat: Adding more fields to the translation

Victorblsilveira committed:
feat: Make Endpoints implement Harvester API OpenAPI specs

jufantozzi committed:
integrity disk

Signed-off-by: William Barrera <wibarre@gmail.com>

Victorblsilveira committed:
feat: Creating modules for generic function pourpouses and organizing adapters function

dependabot[bot] committed:
build(deps): Bump github.com/spiffe/spire-api-sdk from 1.6.2 to 1.6.3

Bumps [github.com/spiffe/spire-api-sdk](https://github.com/spiffe/spire-api-sdk) from 1.6.2 to 1.6.3.
- [Release notes](https://github.com/spiffe/spire-api-sdk/releases)
- [Commits](https://github.com/spiffe/spire-api-sdk/compare/v1.6.2...v1.6.3)

---
updated-dependencies:
- dependency-name: github.com/spiffe/spire-api-sdk
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Victorblsilveira committed:
feat: Fixing error handlers

mgbcaio committed:
Merge pull request #139 from HewlettPackard/dependabot/go_modules/github.com/spiffe/spire-api-sdk-1.6.3

Victorblsilveira committed:
Merge branch 'main' into victor/api-refact

dependabot[bot] committed:
build(deps): Bump github.com/ory/dockertest/v3 from 3.9.1 to 3.10.0

Bumps [github.com/ory/dockertest/v3](https://github.com/ory/dockertest) from 3.9.1 to 3.10.0.
- [Release notes](https://github.com/ory/dockertest/releases)
- [Commits](https://github.com/ory/dockertest/compare/v3.9.1...v3.10.0)

---
updated-dependencies:
- dependency-name: github.com/ory/dockertest/v3
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

mgbcaio committed:
Merge pull request #140 from HewlettPackard/dependabot/go_modules/github.com/ory/dockertest/v3-3.10.0

Victorblsilveira committed:
Merge remote-tracking branch 'origin/main' into victor/api-refact

Victorblsilveira committed:
feat: Changing functions to generic

dependabot[bot] committed:
build(deps): Bump github.com/getkin/kin-openapi from 0.115.0 to 0.116.0

Bumps [github.com/getkin/kin-openapi](https://github.com/getkin/kin-openapi) from 0.115.0 to 0.116.0.
- [Release notes](https://github.com/getkin/kin-openapi/releases)
- [Commits](https://github.com/getkin/kin-openapi/compare/v0.115.0...v0.116.0)

---
updated-dependencies:
- dependency-name: github.com/getkin/kin-openapi
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Victorblsilveira committed:
feat: Adding new definitions in admin handlers

Victorblsilveira committed:
feat: Adjusting response types

dependabot[bot] committed:
build(deps): Bump step-security/harden-runner from 2.3.0 to 2.3.1

Bumps [step-security/harden-runner](https://github.com/step-security/harden-runner) from 2.3.0 to 2.3.1.
- [Release notes](https://github.com/step-security/harden-runner/releases)
- [Commits](https://github.com/step-security/harden-runner/compare/03bee3930647ebbf994244c21ddbc0d4933aab4f...6b3083af2869dc3314a0257a42f4af696cc79ba3)

---
updated-dependencies:
- dependency-name: step-security/harden-runner
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

mchurichi committed:
Merge pull request #141 from HewlettPackard/dependabot/go_modules/github.com/getkin/kin-openapi-0.116.0

build(deps): Bump github.com/getkin/kin-openapi from 0.115.0 to 0.116.0

dependabot[bot] committed:
build(deps): Bump aquasecurity/trivy-action from 0.9.2 to 0.10.0

Bumps [aquasecurity/trivy-action](https://github.com/aquasecurity/trivy-action) from 0.9.2 to 0.10.0.
- [Release notes](https://github.com/aquasecurity/trivy-action/releases)
- [Commits](https://github.com/aquasecurity/trivy-action/compare/1f0aa582c8c8f5f7639610d6d38baddfea4fdcee...e5f43133f6e8736992c9f3c1b3296e24b37e17f2)

---
updated-dependencies:
- dependency-name: aquasecurity/trivy-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Victorblsilveira committed:
Merge pull request #142 from HewlettPackard/dependabot/github_actions/aquasecurity/trivy-action-0.10.0

build(deps): Bump aquasecurity/trivy-action from 0.9.2 to 0.10.0

Victorblsilveira committed:
Merge branch 'main' into dependabot/github_actions/step-security/harden-runner-2.3.1

Victorblsilveira committed:
feat: Spliting AuthN in middleware struct function

mchurichi committed:
Merge pull request #143 from HewlettPackard/dependabot/github_actions/step-security/harden-runner-2.3.1

build(deps): Bump step-security/harden-runner from 2.3.0 to 2.3.1

Victorblsilveira committed:
feat: Minor refacts

Commit:
Update go version (#144)

Signed-off-by: Max Lambrecht <maxlambrecht@gmail.com>

Victorblsilveira committed:
Update pkg/common/api/helper.go

Co-authored-by: Maximiliano Churichi <mchurichi@gmail.com>
Signed-off-by: Victor Vieira Barros Leal da Silveira <victorblsilveira@gmail.com>

Victorblsilveira committed:
fix: removing unused functions

Victorblsilveira committed:
fix: Function call missing name

Victorblsilveira committed:
fix: Validating Trust Domain Input

Victorblsilveira committed:
feat: Undoing generic modifications

Victorblsilveira committed:
Merge branch 'HewlettPackard:main' into victor/api-refact

Victorblsilveira committed:
feat: Updating log

Victorblsilveira committed:
fix: Possible scenario which the token is not found that can cause the auth layer to allow business logic access

Victorblsilveira committed:
feat: Improving error handling

Victorblsilveira committed:
Update pkg/common/http/http.go

Co-authored-by: Max Lambrecht <maxlambrecht@gmail.com>
Signed-off-by: Victor Vieira Barros Leal da Silveira <victorblsilveira@gmail.com>

Victorblsilveira committed:
fix: Simplifying error handling, import names and functions names

jufantozzi commented on pkg/harvester/integrity/repository/disk.go:
```diff
@@ -1 +1,237 @@
+// Configure sets the values of the Disk structure if valid values are present for the paths
+// This function assumes that config parameter has been parsed out based on hcl rules
+func (dc *Disk) Configure(config *Config) error {
+
```
nit: This is code style stuff, but it's not common to add this line break here

jufantozzi commented on pkg/harvester/integrity/repository/disk.go:
```diff
@@ -1 +1,237 @@
+}
+
+type Config struct {
+	CertFilePath           string `hcl:"cert_file_path"`           // Root CA certificate
```
Pattern for these comments (mostly for exported types) is that they should be on top of the declaration and start with the name of the defined variable. This is a standard Go "lightly" enforces and most IDEs leverage that to provide a better way to navigate through the code. 

```
// CertFilePath is the path for Root CA certificate
CertFilePath           string `hcl:"cert_file_path"`

jufantozzi commented on pkg/harvester/integrity/repository/disk.go:
```diff
@@ -1 +1,237 @@
+
+// Disk is a SigningCA that generates signing certificates and validation materials
+// Disk is a local signing CA and has access to root keys and root certs via a path.
+type Disk struct {
```
types that implement an interface (the interface being what's gonna be used by the rest of the code) should be private

jufantozzi commented on pkg/harvester/integrity/repository/disk.go:
```diff
@@ -1 +1,237 @@
+}
+
+func New() (Disk, error) {
+	var disk Disk
+
+	return disk, nil
```
nit: 
```suggestion
	return &disk{}, nil
```

jufantozzi replied:
nit: 
```suggestion
	return &disk{}, nil
```

jufantozzi commented on pkg/harvester/integrity/repository/disk.go:
```diff
@@ -1 +1,237 @@
+type Config struct {
+	CertFilePath           string `hcl:"cert_file_path"`           // Root CA certificate
+	KeyPath                string `hcl:"key_file_path"`            // Root Key
+	ValidationMaterialPath string `hcl:"validation_material_path"` // TrustBundle to validate CA signed material
```
I don't know if I'm missing something but I don't know how exactly we would be using this validation material. I sketched this diagram to clarify how I'm seeing the chain of trust, see if that makes sense to you.

![image](https://user-images.githubusercontent.com/20044948/234725296-43750d5f-fbe0-481f-92b6-c2ca3a114df3.png)

This means that all we need to return from this layer is a single public key that was used to sign the certificates issued to harvesters. Am I missing something here?

Victorblsilveira committed:
feat: Implementing unit test for helper

Victorblsilveira committed:
feat: Including unit tests for TCP and UDS handlers

Victorblsilveira committed:
feat: Introducing in memory database and including in auth unit tests

Victorblsilveira committed:
fix: Correcting auth unit test assertions

Commit:
Add X509CA interface and disk implementation (#145)

Add X509CA interface and disk implementation

Signed-off-by: Max Lambrecht <maxlambrecht@gmail.com>

jufantozzi commented on pkg/harvester/integrity/repository/disk.go:
```diff
@@ -1 +1,238 @@
+	//rootCertificate is the  root certificate of the signing CA
+	rootCertificate *x509.Certificate
+	//validationBundle is the list of certificates to be used for validation of signing certificates
+	validationBundle []*x509.Certificate
```
I thought we were dropping this?

jufantozzi replied:
unless we're generating an "intermediate" from this root we're reading (and that would be the validation bundle), but i'd assume that "root" to be an intermediate itself already, provided by whatever is supplying this

wibarre replied:
I was thinking and I believe we will need this for other implementations. If we were to have the GCA, we would need a way to retrieve and make available validation material (or bundle validation) for the signing certs. For Disk, it is the same root cert, but for other implementations that could be more complexity.

Victorblsilveira committed:
feat: Removing error types

Victorblsilveira committed:
feat: Implementing unit test for some handlers

Victorblsilveira committed:
fix: Unit test error checking

Victorblsilveira committed:
feat: Implementing other fake functions in fakedatabase

Victorblsilveira committed:
Update pkg/server/datastore/fakedatabase.go

Co-authored-by: Max Lambrecht <maxlambrecht@gmail.com>
Signed-off-by: Victor Vieira Barros Leal da Silveira <victorblsilveira@gmail.com>

Victorblsilveira committed:
Update pkg/server/endpoints/run.go

Co-authored-by: Max Lambrecht <maxlambrecht@gmail.com>
Signed-off-by: Victor Vieira Barros Leal da Silveira <victorblsilveira@gmail.com>

Victorblsilveira committed:
Update pkg/server/endpoints/run.go

Co-authored-by: Max Lambrecht <maxlambrecht@gmail.com>
Signed-off-by: Victor Vieira Barros Leal da Silveira <victorblsilveira@gmail.com>

Victorblsilveira committed:
Update pkg/server/datastore/fakedatabase.go

Co-authored-by: Max Lambrecht <maxlambrecht@gmail.com>
Signed-off-by: Victor Vieira Barros Leal da Silveira <victorblsilveira@gmail.com>

Victorblsilveira committed:
Update pkg/server/datastore/fakedatabase.go

Co-authored-by: Max Lambrecht <maxlambrecht@gmail.com>
Signed-off-by: Victor Vieira Barros Leal da Silveira <victorblsilveira@gmail.com>

Victorblsilveira committed:
fix: Addressing some renaming and misstypings

Victorblsilveira committed:
Merge branch 'main' into victor/api-refact

Victorblsilveira committed:
fix: Adding some comments and error checking

Victorblsilveira committed:
fix: Checking error by contains

Victorblsilveira committed:
fix: Asserting Error with trust domain name

Victorblsilveira committed:
fix: Management API documentation

Commit:
Add TLS communication between Galadriel Server and Harvester (#146)

Add TLS communication between Server and Harvester

Signed-off-by: Max Lambrecht <maxlambrecht@gmail.com>

Victorblsilveira committed:
Merge remote-tracking branch 'origin/main' into victor/api-refact

Victorblsilveira committed:
Merge pull request #138 from Victorblsilveira/victor/api-refact

Refactoring Models to use OpenAPI definitions

dependabot[bot] committed:
build(deps): Bump google.golang.org/grpc from 1.54.0 to 1.55.0

Bumps [google.golang.org/grpc](https://github.com/grpc/grpc-go) from 1.54.0 to 1.55.0.
- [Release notes](https://github.com/grpc/grpc-go/releases)
- [Commits](https://github.com/grpc/grpc-go/compare/v1.54.0...v1.55.0)

---
updated-dependencies:
- dependency-name: google.golang.org/grpc
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): Bump step-security/harden-runner from 2.3.1 to 2.4.0

Bumps [step-security/harden-runner](https://github.com/step-security/harden-runner) from 2.3.1 to 2.4.0.
- [Release notes](https://github.com/step-security/harden-runner/releases)
- [Commits](https://github.com/step-security/harden-runner/compare/6b3083af2869dc3314a0257a42f4af696cc79ba3...128a63446a954579617e875aaab7d2978154e969)

---
updated-dependencies:
- dependency-name: step-security/harden-runner
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Victorblsilveira committed:
Merge pull request #149 from HewlettPackard/dependabot/github_actions/step-security/harden-runner-2.4.0

build(deps): Bump step-security/harden-runner from 2.3.1 to 2.4.0

Victorblsilveira committed:
Merge branch 'main' into dependabot/go_modules/google.golang.org/grpc-1.55.0

Victorblsilveira committed:
Merge pull request #148 from HewlettPackard/dependabot/go_modules/google.golang.org/grpc-1.55.0

build(deps): Bump google.golang.org/grpc from 1.54.0 to 1.55.0

dependabot[bot] committed:
build(deps): Bump golang.org/x/crypto from 0.8.0 to 0.9.0

Bumps [golang.org/x/crypto](https://github.com/golang/crypto) from 0.8.0 to 0.9.0.
- [Commits](https://github.com/golang/crypto/compare/v0.8.0...v0.9.0)

---
updated-dependencies:
- dependency-name: golang.org/x/crypto
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

mgbcaio committed:
Merge pull request #152 from HewlettPackard/dependabot/go_modules/golang.org/x/crypto-0.9.0

Commit:
Add JWT issuing and validation. Change Harvester onboarding and authentication processes. (#151)

Add JWT issuing and validation
Change Harvester onboarding and authentication processes.

Signed-off-by: Max Lambrecht <maxlambrecht@gmail.com>

wibarre committed:
integrity disk

Signed-off-by: William Barrera <wibarre@gmail.com>

Commit:
Merge branch 'main' into integrity/disk
